engine/src/main/java/org/terasology/rendering/nui/asset/UIFormat.java

```diff
@@ -196,8 +160,20 @@ public UIWidget deserialize(JsonElement json, Type typeOfT, JsonDeserializationC
             }
 
             // Deserialize normal fields.
+            Set<String> unknownFields = new HashSet<>();
+            for (Entry<String, JsonElement> entry : jsonObject.entrySet()) {
+                String name = entry.getKey();
+                if (!ID_FIELD.equals(name)
```

Comment:
Would consider putting these fields in a static immutable set or similar, and replacing with a call to contains(), but not a big deal with only 4 fields.